test/sqllogictest/tpch.slt

```diff
@@ -241,41 +237,35 @@ ORDER BY
 7 =
 | Get %5
 | Distinct group=(#0)
-| ArrangeBy (#0)
+| ArrangeBy ()
 
 8 =
 | Get materialize.public.partsupp (u11)
-| ArrangeBy (#0) (#1)
+| ArrangeBy (#0)
 
 9 =
 | Get materialize.public.supplier (u8)
-| ArrangeBy (#0) (#3)
+| ArrangeBy (#0, #3)
 
 10 =
 | Get materialize.public.nation (u1)
-| ArrangeBy (#0) (#2)
+| ArrangeBy ()
 
 11 =
 | Get materialize.public.region (u4)
-| ArrangeBy (#0)
+| Filter (#1 = "EUROPE")
 
 12 =
-| Join %7 %8 %9 %10 %11 (= %7.#0 %8.#0) (= %8.#1 %9.#0) (= %9.#3 %10.#0) (= %10.#2 %11.#0)
-| | implementation = DeltaQuery %7 %8.(#0) %9.(#0) %10.(#0) %11.(#0) | %8 %7.(#0) %9.(#0) %10.(#0) %11.(#0) | %9 %10.(#0) %11.(#0) %8.(#1) %7.(#0) | %10 %11.(#0) %9.(#3) %8.(#1) %7.(#0) | %11 %10.(#2) %9.(#3) %8.(#1) %7.(#0)
-| | demand for %7 = (#0)
-| | demand for %8 = (#3)
-| | demand for %9 = ()
-| | demand for %10 = ()
-| | demand for %11 = (#1)
-| Filter (#18 = "EUROPE")
+| Join %7 %8 %9 %10 %11 (= #0 #1 #9) (= #2 #6) (= #14 #16)
+| | implementation = Differential %11 %7.() %8.(#0) %9.(#0, #3) %10.()
```

Comment:
I'll investigate! The filter appears to have been pushed down, and if that happened first, then perhaps that is the source of the trouble. Await further details!

Reply:
Another thing that has changed is that there are fewer constraints, one with three terms rather than all with two. That seems more exciting! Still investigating.

Reply:
Yeah, that appears to be `s_nationkey` equated with a bunch of non-nationkey things. Part keys, it looks like. Some bug for sure. Investigating!

Reply:
Looks like a bug in join fusion. Fixing!

Reply:
Resolution was: actually a bug in `redundant_join` where is mis-updated column identifiers.